routing/routing.xml

```diff
@@ -23,29 +23,238 @@
+    
+        Priority is a preference multiplicator (between 0 and 1), which can only be used to _de_prioritize routes.
+        The formula used is 
+        `minimum(speed * multiplier, maxDefaultSpeed)`; where maxDefaultSpeed is the maximum allowed speed on the route; and speed is the earlier calculated speed
+
+
+    ## 5) Obstacle (specified on points, penalty time in seconds or -1 if blocked)
+    
+        This block evaluates points encountered on the road.
+        The block specifies if this obstacle can be passed and eventually a penalty for passing it.
+        
+        A value of "-1" is given if the obstacle can not be passed;
+        A value of "0" is given if the obstacle can be passed without problem.
+        A penalty in seconds is given if the obstacle slows you down. (E.g. a kerb might slow dow 
+        
+        This value is only used by the routing engine
+        
+
+        E.g. a kerb is a showstopper for a wheelchair
+        <routingProfile name="wheelchair">
+            <point attribute="obstacle>
+                <select value="-1" t="barrier" v="kerb" />
+            </point>
+        </routingProfile>    
+        
+        
+        E.g. a kerb causes a slight delay for cyclists
+        <routingProfile name="bicycle">
+            <point attribute="obstacle>
+                <select value="5" t="barrier" v="kerb" />
+            </point>
+        </routingProfile>  
+        
+
+    ## 6) Obstacle_time (specified on points, penalty time in seconds)
+
+        Gives a slowdown time for an obstacle in seconds.
+        In contract to `obstacle`, this value is only used to calculate an ETA and show it on the UI once the route is calculated.
+        In other words, the values here do not influence routing.
+        
+        
+
+    ## 7) Penalty_transition (specified on ways, penalty time in seconds)
+
+        When driving a car, changing from a motorway to a trunk slows you down: you should already slow down 500 meters before the trunk.
+        This has a penalty which is taken into account by (only) the navigation algorithm.
+        This is done by taking the difference between two penalties.
+        The formula used is `maximum(0, penalty_new_way - penalty_current_way)`.
+        
+        E.g. 
+        <way attribute="penalty_transition">
+            <select value="0" t="highway" v="motorway"/>
+            <select value="5" t="highway" v="motorway_link"/>
+            <select value="20" t="highway" v="tertiary"/>
+        </way>
+        
+        
+        This means that, going from a `motorway_link` to a `tertiary`, the penalty given will be
+        `maximum(0, 20 - 5)` or `15 seconds`.
+        
+        On the other hand, going from motorway to motorway incurs no penalty, neither does going from a motorway_link to a motorway.
+
+    ## 8) obstacle_srtm_alt_speed (specified on ways, in seconds of penalty)
+    
+	What the fuck does this block?
```

Comment:
Meaning of select value:
penalty in seconds for every height metre

For example:

>				<gt value1="-60" value2=":incline">
>					<select value="35"/>
>				</gt>
>				<gt value1="-25" value2=":incline">
>					<select value="30"/>
>				</gt>
>				<gt value1="-17" value2=":incline">
>					<select value="17"/>
>				</gt>
>				<gt value1="-9" value2=":incline">
>					<select value="5"/>
>				</gt>
>				<gt value1="-3" value2=":incline">
>					<select value="1"/>
>				</gt>
>				<gt value1="3" value2=":incline">
>					<select value="1"/>
>				</gt>
>				<gt value1="7" value2=":incline">
>					<select value="4"/>
>				</gt>
>				<gt value1="13" value2=":incline">
>					<select value="8"/>
>				</gt>
>				<gt value1="25" value2=":incline">
>					<select value="10"/>
>				</gt>
>				<select value="15"/>

has the following meaning:

more than 25% => 15 seconds per height metre
[25% 13%) => 10 seconds per height metre
[13% 7%) => 8 seconds per height metre
[7% 3%) => 4 seconds per height metre
[3% -3%) => 1 seconds per height metre
[-3% -9%) => 1 seconds per height metre
[-9% -17%) => 5 seconds per height metre
[-17% -25%) => 17 seconds per height metre
[25% -60%) => 30 seconds per height metre
60% and less => 35 seconds per height metre

negative select value (seconds per height metre) means "impassable"

Let the incline of a part of a way be 20%, then the penalty per height metre is 10 seconds.
If the part of the way has 9 height meters, the penalty for this part will be 10*9 = 90 seconds.

The height penalty is additional.
Total routing engine time is:
height penalty + distance/(speed*prio) + obstacles + turn penalties + ....

Reply:
I copied your explanation and cleaned it up a little. New commit coming soon.